lib/also_energy/site.rb

```diff
@@ -0,0 +1,13 @@
+module AlsoEnergy
+  class Site
+    include HashWrangler
+    include ReturnAttributes
+
+    attr_accessor :id, :name, :latitude, :longitude, :city, :state, :time_zone, :dst, :type
+
+    def initialize
+      yield(self) if block_given?
+    end
+
+  end
+end
```

Comment:
I would recommend using Virtus here. Virtus will let you define attributes, and will coerce them into the types that you define. It would look something like this:

```ruby
module AlsoEnergy
  class Site
    include Virtus.model

    attribute :id, Integer
    attribute :name, String
    attribute :latitude, Float
    attribute :longitude, Float
    attribute :city, String
    attribute :state, String
    attribute :time_zone, String
    attribute :dst, Boolean
    attribute :type, String

    def initialize(params = {})
      super(params)
    end
  end
end
```

Then you can just pass a hash of parameters into the `.new` method and the super call will pass the params to virtus, and it will generate attr_accessor methods for you, and will coerce each value to the type that you specify. You also get an `.attributes` method for free that will serialize all of the attributes into a hash, much like ActiveRecord does. 

This will also make your get_site_list method much simpler. You can change the creation of the array of sites to this:
```ruby
def get_sites
    message = { 'als:sessionID' => session_id }
    response = find_in_hash(:items, (connection.call(:get_site_list, message: message).body))
    fail QueryError, "Query Failed!" if response == nil
    response.map { |site| AlsoEnergy::Site.new(site) }
end
```